core/src/main/java/org/opensearch/sql/datasource/DataSourceServiceImpl.java

```diff
@@ -95,12 +112,28 @@ public void createDataSource(DataSourceMetadata metadata) {
 
   @Override
   public void updateDataSource(DataSourceMetadata dataSourceMetadata) {
-    throw new UnsupportedOperationException("will be supported in future");
+    validateDataSourceMetaData(dataSourceMetadata);
+    if (!dataSourceMetadata.getName().equals(DEFAULT_DATASOURCE_NAME)) {
+      this.dataSourceMetadataStorage.updateDataSourceMetadata(dataSourceMetadata);
+      clearDataSourceWithSameName(dataSourceMetadata.getName());
+      dataSourceMap.put(dataSourceMetadata,
+          dataSourceFactoryMap.get(dataSourceMetadata.getConnector())
+              .createDataSource(dataSourceMetadata));
```

Comment:
does validateDataSourceMetaData required? should createDataSource also validate datasource?

Reply:
what is create datasource failed, should we clear datasource when create successsfully?